src/prefect/client/orion.py

```diff
@@ -161,6 +160,18 @@ def __init__(
         self._client = PrefectHttpxClient(
             **httpx_settings,
         )
+
+        # See https://www.python-httpx.org/advanced/#custom-transports
+        # If we're using an HTTP/S client (not the ephemeral client), adjust the
+        # transport to add retries _after_ it is instantiated. If we alter the transport
+        # before instantiation, the transport will not be aware of proxies unless we
+        # reproduce all of the logic to make it so.
+        if isinstance(api, str):
+            orion_transport = self._client._transport_for_url(httpx.URL(api))
+            if isinstance(orion_transport, httpx.AsyncHTTPTransport):
+                if isinstance(orion_transport._pool, httpcore.AsyncConnectionPool):
+                    orion_transport._pool._retries = 3
```

Comment:
Since we were using `setdefault` before this no longer lets a user pass in a transport without our retry count. I don't think that's a big deal, but I'm also not excited that we're using all these private attributes. Is there no better way to do this?

Reply:
Ahh that's a great catch, I can skip this if the users provides a transport themselves.

As far as alternatives, it's looked pretty rough to me (and seems a lot worse than poking into the transport's pool to set the retries).  I believe we'd have to reproduce what's happening here:

https://github.com/encode/httpx/blob/master/httpx/_client.py#L213-L231
https://github.com/encode/httpx/blob/master/httpx/_client.py#L689-L702

I feel like with the `isinstance` checks and the unit tests, we're protecting ourselves pretty well against changes in the library.

I welcome alternatives, though!  We'll need to have solid support for proxies in many enterprise, government, and academic settings, so it's well worth it.

Reply:
Hm. The httpx proxy logic is complicated because its for a client that can contact many hosts. Our client is only being used for one host though and we know what it is when we start the client. Have you looked at https://github.com/PrefectHQ/prefect/issues/7204?

How are we configuring proxies in Cloud for this? `HTTPS_PROXY`?

> I feel like with the isinstance checks and the unit tests, we're protecting ourselves pretty well against changes in the library.

I'm more worried about compatibility breaking in released versions of Prefect, e.g. we release this then httpx changes the name of an attribute in a minor version a few months later. Users of the old version of Prefect will have to manually pin their httpx lower or they won't be able to use any server.

> We'll need to have solid support for proxies in many enterprise, government, and academic settings, so it's well worth it.

I strongly agree :)

Reply:
After a synchronous discussion, we'll add some more guards here but this looks like the best path forward. We'll also be reaching out to the encode/httpx team to see if we can contribute upstream to avoid this pattern.